### DISCUSSION TITLE
How to use the Rule Engine "MQTT Subscription" Resource?

### DISCUSSION THREAD

**Discussion by matthewGEA** [emqx/emqx#4587] [Q&A]
Using Broker version 4.2.9
I have created a connection from EMQx broker to another EMQx broker using the MQTT Subscription (bridge_mqtt_sub)

![image](https://user-images.githubusercontent.com/46966648/114964267-5ec30e80-9e3c-11eb-956e-53a64d3c23ae.png)

I used a POOL =2 and I see QTY 2 client connections listed on the remote broker.  Neither of these clients have subscriptions.

What needs to be done to use the MQTT Subscription Resource in the Rules Engine?

**Comment by terry-xiaoyu** [comment]
Could you enable the debug logs of both the log and remote brokers by `emqx_ctl log set-level debug`, and then re-create the subscription resource again, too see what are the logs?

**Reply by matthewGEA** [reply]
The local broker is a 2-node cluster running 4.2.9.  The remote broker is 4.3-rc.3.  The connection was setup for POOL=2.  Subscription_opts is [ { "topic": "spBv1.0/#", "qos": 0 } ]
Below is what I see on the bridge (local) broker:

> (emqx@host.localdomain)1> 2021-04-16 12:54:12.403 [info] Initiating Resource bridge_mqtt, ResId: <<"resource_mqtt_subscribe">>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.403 [info] Initiating Resource bridge_mqtt, ResId: <<"resource_mqtt_subscribe">>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.409 [debug] emqtt(client_5C0167A0356FFC0E9000009BC0000): SEND Data: {mqtt_packet,{mqtt_packet_header,1,false,0,false},{mqtt_packet_connect,<<"MQTT">>,4,true,true,false,0,false,60,#{},<<"client_5C0167A0356FFC0E9000009BC0000">>,undefined,undefined,undefined,<<>>,<<>>},undefined}
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.410 [info] Initiated Resource bridge_mqtt Successfully, ResId: <<"resource_mqtt_subscribe">>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.410 [debug] emqtt(client_5C0167A035BE6C0E9000009BF0000): SEND Data: {mqtt_packet,{mqtt_packet_header,1,false,0,false},{mqtt_packet_connect,<<"MQTT">>,4,true,true,false,0,false,60,#{},<<"client_5C0167A035BE6C0E9000009BF0000">>,undefined,undefined,undefined,<<>>,<<>>},undefined}
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.409 [info] Initiated Resource bridge_mqtt Successfully, ResId: <<"resource_mqtt_subscribe">>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.412 [debug] emqtt(client_5C0167A0356FFC0E9000009BC0000): RECV Data: <<32,2,0,0>>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.413 [info] [Bridge] Bridge 'emqx_bridge_worker_bridge_mqtt:resource_mqtt_subscribe:1' is connecting......
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.413 [debug] emqtt(client_5C0167A035BE6C0E9000009BF0000): RECV Data: <<32,2,0,0>>
> (emqx@host.localdomain)1> 2021-04-16 12:54:12.413 [info] [Bridge] Bridge 'emqx_bridge_worker_bridge_mqtt:resource_mqtt_subscribe:2' is connecting......



Below is what I see on the remote broker:

> 2021-04-16 12:54:22.431 [debug] 10.22.79.10:37661 [MQTT] RECV <<16,52,0,4,77,81,84,84,132,194,0,60,0,36,99,108,105,101,110,116,95,53,67,48,49,54,55,65,48,51,53,55,69,67,52,54,48,70,48,48,48,48,48,57,68,66,48,48,48,48,0,0,0,0>>
> 2021-04-16 12:54:22.431 [debug] 10.22.79.10:37661 [MQTT] RECV CONNECT(Q0, R0, D0, ClientId=client_5C0167A0357EC460F000009DB0000, ProtoName=MQTT, ProtoVsn=4, CleanStart=true, KeepAlive=60, Username=, Password=******)
> 2021-04-16 12:54:22.431 [debug] 10.22.79.10:37687 [MQTT] RECV <<16,52,0,4,77,81,84,84,132,194,0,60,0,36,99,108,105,101,110,116,95,53,67,48,49,54,55,65,48,51,53,54,70,70,67,48,69,57,48,48,48,48,48,57,66,67,48,48,48,48,0,0,0,0>>
> 2021-04-16 12:54:22.431 [debug] 10.22.79.10:37687 [MQTT] RECV CONNECT(Q0, R0, D0, ClientId=client_5C0167A0356FFC0E9000009BC0000, ProtoName=MQTT, ProtoVsn=4, CleanStart=true, KeepAlive=60, Username=, Password=******)
> 2021-04-16 12:54:22.432 [debug] <<"client_5C0167A0356FFC0E9000009BC0000">>@10.22.79.10:37687 [MQTT] SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
> 2021-04-16 12:54:22.432 [debug] <<"client_5C0167A0357EC460F000009DB0000">>@10.22.79.10:37661 [MQTT] SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
> 2021-04-16 12:54:22.432 [debug] 10.22.79.10:34375 [MQTT] RECV <<16,52,0,4,77,81,84,84,132,194,0,60,0,36,99,108,105,101,110,116,95,53,67,48,49,54,55,65,48,51,53,66,69,54,67,48,69,57,48,48,48,48,48,57,66,70,48,48,48,48,0,0,0,0>>
> 2021-04-16 12:54:22.432 [debug] 10.22.79.10:34375 [MQTT] RECV CONNECT(Q0, R0, D0, ClientId=client_5C0167A035BE6C0E9000009BF0000, ProtoName=MQTT, ProtoVsn=4, CleanStart=true, KeepAlive=60, Username=, Password=******)
> 2021-04-16 12:54:22.433 [debug] <<"client_5C0167A035BE6C0E9000009BF0000">>@10.22.79.10:34375 [MQTT] SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
> 2021-04-16 12:54:22.433 [debug] 10.22.79.10:34053 [MQTT] RECV <<16,52,0,4,77,81,84,84,132,194,0,60,0,36,99,108,105,101,110,116,95,53,67,48,49,54,55,65,48,51,53,67,55,53,52,54,48,70,48,48,48,48,48,57,68,69,48,48,48,48,0,0,0,0>>
> 2021-04-16 12:54:22.433 [debug] 10.22.79.10:34053 [MQTT] RECV CONNECT(Q0, R0, D0, ClientId=client_5C0167A035C75460F000009DE0000, ProtoName=MQTT, ProtoVsn=4, CleanStart=true, KeepAlive=60, Username=, Password=******)
> 2021-04-16 12:54:22.433 [debug] <<"client_5C0167A035C75460F000009DE0000">>@10.22.79.10:34053 [MQTT] SEND CONNACK(Q0, R0, D0, AckFlags=0, ReasonCode=0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0357EC460F000009DB0000">>@10.22.79.10:37661 [MQTT] RECV <<192,0>>
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0357EC460F000009DB0000">>@10.22.79.10:37661 [MQTT] RECV PINGREQ(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0357EC460F000009DB0000">>@10.22.79.10:37661 [MQTT] SEND PINGRESP(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0356FFC0E9000009BC0000">>@10.22.79.10:37687 [MQTT] RECV <<192,0>>
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0356FFC0E9000009BC0000">>@10.22.79.10:37687 [MQTT] RECV PINGREQ(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A0356FFC0E9000009BC0000">>@10.22.79.10:37687 [MQTT] SEND PINGRESP(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A035BE6C0E9000009BF0000">>@10.22.79.10:34375 [MQTT] RECV <<192,0>>
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A035BE6C0E9000009BF0000">>@10.22.79.10:34375 [MQTT] RECV PINGREQ(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A035BE6C0E9000009BF0000">>@10.22.79.10:34375 [MQTT] SEND PINGRESP(Q0, R0, D0)
> 2021-04-16 12:55:22.449 [debug] <<"client_5C0167A035C75460F000009DE0000">>@10.22.79.10:34053 [MQTT] RECV <<192,0>>
> 2021-04-16 12:55:22.450 [debug] <<"client_5C0167A035C75460F000009DE0000">>@10.22.79.10:34053 [MQTT] RECV PINGREQ(Q0, R0, D0)
> 2021-04-16 12:55:22.450 [debug] <<"client_5C0167A035C75460F000009DE0000">>@10.22.79.10:34053 [MQTT] SEND PINGRESP(Q0, R0, D0)

**Comment by matthewGEA** [comment]
Reference here https://github.com/emqx/emqx/issues/4621

**Comment by Trance-Paradox** [comment]
Can you list the steps how you did it?
I am not able to find any plugin named bridge_mqtt_sub
